Lib/asyncio/base_events.py

```diff
@@ -560,7 +565,12 @@ def _asyncgen_firstiter_hook(self, agen):
         try:
             await future
         finally:
-            thread.join()
+            thread.join(timeout)
+
+        if (thread.is_alive()):
+            warnings.Warning("The ThreadPoolExecutor did not finishing joining"
```

Comment:
It it `Warning` or `warning`?

Reply:
@1st1 I believe `Warning` is correct https://docs.python.org/3/library/exceptions.html#Warning.

Reply:
Well, yes, you construct a Warning instance, but you don't send it anywhere. Usually we use the https://docs.python.org/3/library/warnings.html#warnings.warn function or similar.

Reply:
@1st1

Oops. Thanks for the clarification, I was thinking along the lines of ``raise Warning(msg)" and somehow ended up with ``warnings.Warning(msg)``. Using `warnings.warn()` makes far more sense. I know I saw it used multiple times while I was reverting the 3.8 API, but I got the syntax mixed up in my head. Warnings are definitely not something that I've utilized in my own code regularly, I typically stick to exceptions or logging. 

I have no idea how I managed to get the two mixed up though, that's a tad bit embarrassing.

Reply:
@1st1 Couple of questions regarding the warning:

1) Should it be a `RuntimeWarning`? I think this would be suitable since it occurs during runtime as a result of the thread not being able to close within the specified timeout window. I could imagine an argument for it being a `ResourceWarning` as well though.

2) Should I add a test to `test_asyncio/test_events.py` to make sure the the warning is triggered in an appropriate scenario? I have some familiarity with the existing tests, but it would be my first time writing a new one.